puppet/modules/quickstack/manifests/pacemaker/redis.pp

```diff
@@ -0,0 +1,40 @@
+class quickstack::pacemaker::redis(
+  $bind_host             = '127.0.0.1',
+  $port                  = '6379',
+  $slaveof               = undef,
+) {
+
+  $redis_group = map_params("redis_group")
+
+  class {'::quickstack::firewall::redis':
+    ports => [$port],
+  }
+
+  class {'::quickstack::db::redis':
+    bind_host => $bind_host,
+    port      => $port,
+    slaveof   => $slaveof,
+  }
+
+  quickstack::pacemaker::resource::generic {'redis':
+    clone_opts    => "interleave=true",
+    operation_opts => "wait_last_known_master=true notify=true ordered=true"
+    resource_name => "redis",
+    resource_params => "--master meta",
+  } ->
+
+  quickstack::pacemaker::vips { $redis_group:
+      public_vip   => map_params('redis_vip'),
+      private_vip  => map_params('redis_vip'),
+      admin_vip    => map_params('redis_vip'),
```

Comment:
Does redis need a public, private and admin VIP?

I had thought this pattern was inspired by the public/private/admin URLs registered for an endpoint in the keystone service catalog, which would only generally be relevant to the openstack API services (as opposed to a backend service like redis).